### PR TITLE
Use Test262Harness to run tests

### PR DESCRIPTION
**Pull request by lahma**
* tests loaded from test262 repository
* test cases generated from test files
* settings file used to know what to ignore/include during generation
* deleted EcmaTest.csproj so they won't be run anymore, separate *BIG* delete commit required for deleting test files in VCS

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <clear />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
     <add key="Esprima" value="https://www.myget.org/F/esprimadotnet/api/v3/index.json" />
+    <add key="Local" value="libs" />
```
Comment (sebastienros) [inline NuGet.config:7]: left-over?

File: Jint/Native/Date/DateConstructor.cs [hunk 2]
```diff
@@ -87,9 +87,26 @@ protected override void Initialize()
             SetProperties(properties);
         }
 
+        /// <summary>
+        /// https://tc39.es/ecma262/#sec-date.parse
+        /// </summary>
         private JsValue Parse(JsValue thisObj, JsValue[] arguments)
         {
             var date = TypeConverter.ToString(arguments.At(0));
+            var negative = date.StartsWith("-");
```
Comment (sebastienros) [inline Jint/Native/Date/DateConstructor.cs:96]: wtf

File: Jint.Tests.Test262/TestHarness.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        // NOTE: The Date tests in test262 assume the local timezone is Pacific Standard Time
+        try
+        {
+            State.TimeZone = TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time");
+        }
+        catch (TimeZoneNotFoundException)
+        {
+            // https://stackoverflow.com/questions/47848111/how-should-i-fetch-timezoneinfo-in-a-platform-agnostic-way
+            // should be natively supported soon https://github.com/dotnet/runtime/issues/18644
```
Comment (sebastienros) [inline Jint.Tests.Test262/TestHarness.cs:24]: It's a lie. It depends on how Windows is configured, so for instance it won't work on Azure App Service.

**Review by sebastienros — APPROVED** [review]

File: Jint.Tests.Test262/TestHarness.cs (hunk 3 above)
Comment (lahma) [reply Jint.Tests.Test262/TestHarness.cs:24]: Good point, I think this is "old code".

File: Jint/Native/Date/DateConstructor.cs (hunk 2 above)
Comment (lahma) [reply Jint/Native/Date/DateConstructor.cs:96]: Trial error horror I guess, I will be having only things related to test in this PR when merged, will exclude Jint changes.

File: NuGet.config (hunk 1 above)
Comment (lahma) [reply NuGet.config:7]: Still needed for NUnit local lib

**Comment by lahma** [comment]
I'll try to port fixes to main to reduce failures, I'd just hate to create a large todo list by having skipped test cases which fail in new tests, but the amount of failures is a bit overwhelming to fix..

File: NuGet.config (hunk 1 above)
Comment (lahma) [reply NuGet.config]: No longer needed, using nuget.org package now

**Comment by lahma** [comment]
/cc recent code contributors @christianrondeau @Jither @KurtGokhan @source-transformer @shlomo-ildar 

You now need to generate Test262 test suite in order for the solution to compile. See [Jint.Tests.Test262](https://github.com/sebastienros/jint/tree/main/Jint.Tests.Test262) project's README.

**Pull request merged**